### PR TITLE
fix: hide uniswap info if non-ethereum network was selected

### PR DESCRIPTION
**Pull request by guanbinrui**
closes #

### CODE REVIEW

File: packages/maskbook/src/plugins/Trader/apis/trending/index.ts [hunk 1]
```diff
@@ -175,13 +176,17 @@ export async function checkAvailabilityOnDataProvider(keyword: string, type: Tag
...
+            )
+            .map(
+                async (x) =>
+                    [
+                        x.value,
+                        await checkAvailabilityOnDataProvider(resolveAlias(keyword, x.value), type, x.value),
+                    ] as const,
+            ),
     )
     return checked.filter(([_, y]) => y).map(([x]) => x)
```
Comment (Himself65) [inline packages/maskbook/src/plugins/Trader/apis/trending/index.ts:191]:
```suggestion
    return checked.filter(([, y]) => y).map(([x]) => x)
```

**Review by Himself65 — APPROVED** [review]
LGTM

**Review by Jack-Works — APPROVED** [review]

File: packages/maskbook/src/plugins/Trader/apis/trending/index.ts [hunk 2]
```diff
@@ -175,15 +176,19 @@ export async function checkAvailabilityOnDataProvider(keyword: string, type: Tag
...
+            .map(
+                async (x) =>
+                    [
+                        x.value,
+                        await checkAvailabilityOnDataProvider(resolveAlias(keyword, x.value), type, x.value),
+                    ] as const,
+            ),
     )
-    return checked.filter(([_, y]) => y).map(([x]) => x)
+    return checked.filter(([, y]) => y).map(([x]) => x)
```
Comment (UncleBill) [inline packages/maskbook/src/plugins/Trader/apis/trending/index.ts:191]: just want the logic more clear

```diff
diff --git a/packages/maskbook/src/plugins/Trader/apis/trending/index.ts b/packages/maskbook/src/plugins/Trader/apis/trending/index.ts
index 7a3af37e2..56fe645a8 100644
--- a/packages/maskbook/src/plugins/Trader/apis/trending/index.ts
+++ b/packages/maskbook/src/plugins/Trader/apis/trending/index.ts
@@ -18,7 +18,6 @@ import {
 import { NetworkType } from '@dimensiondev/web3-shared'
 import { currentNetworkSettings } from '../../../Wallet/settings'
 
-
 /**
  * Get supported currencies of specific data provider
  * @param dataProvider
@@ -176,19 +175,15 @@ export async function checkAvailabilityOnDataProvider(keyword: string, type: Tag
 
 export async function getAvailableDataProviders(type: TagType, keyword: string) {
     const checked = await Promise.all(
-        getEnumAsArray(DataProvider)
-            .filter((x) =>
-                x.value === DataProvider.UNISWAP_INFO ? currentNetworkSettings.value === NetworkType.Ethereum : true,
-            )
-            .map(
-                async (x) =>
-                    [
-                        x.value,
-                        await checkAvailabilityOnDataProvider(resolveAlias(keyword, x.value), type, x.value),
-                    ] as const,
-            ),
+        getEnumAsArray(DataProvider).reduce((promises: Promise<boolean>[], provider) => {
+            if (provider.value === DataProvider.UNISWAP_INFO && currentNetworkSettings.value !== NetworkType.Ethereum) {
+                return promises
+            }
+            promises.push(checkAvailabilityOnDataProvider(resolveAlias(keyword, provider.value), type, provider.value))
+            return promises
+        }, []),
     )
-    return checked.filter(([, y]) => y).map(([x]) => x)
+    return checked.filter(Boolean)
 }
 
 export async function getAvailableCoins(keyword: string, type: TagType, dataProvider: DataProvider) {
```

Comment (guanbinrui) [reply packages/maskbook/src/plugins/Trader/apis/trending/index.ts]: ACCEPTED

**Comment by UncleBill** [comment]
Oops, looks like my patch gets reverted.

**Pull request merged**